kubetest/main.go

```diff
@@ -389,10 +389,16 @@ func complete(o *options) error {
 	return nil
 }
 
-func acquireKubernetes(o *options) error {
+func acquireKubernetes(o *options, d deployer) error {
 	// Potentially build kubernetes
 	if o.build.Enabled() {
-		err := control.XMLWrap(&suite, "Build", o.build.Build)
+		var err error
+		// kind deployer manages build
```

Comment:
this is a bit /shrug, but it makes kubetest do the right thing without redundant builds etc.

Reply:
no idea what this is doing. /shrug

Reply:
So this is the part where kubetest actually takes action for `--build`, this `control.XMLWrap` call will call the function provided as the last arg and based on if it returns nil or an error it will write a junit entry for `"Build`. 

The change is that we made it so that when `--deployment=kind` is set, it will call kind's build, which also includes building the node image, copying around `kubectl`, etc.
When `--deployment=kind` is not set, it will call the usual function that does EG `make quick-release` etc...